### ISSUE TITLE
`disasm-test`: Run some tests only on certain LLVM versions

### ISSUE THREAD

**Issue by RyanGlScott** [GaloisInc/llvm-pretty-bc-parser#206]
Later LLVM versions often add new features to the bitcode language that cannot be parsed by earlier LLVM releases. We would still like to test these new features without having old versions of LLVM break when running `disasm-test`. To make this possible, we should institute a convention where files with certain suffixes—e.g., `mytest.llvm13.ll`—are only run with LLVM 13 or later, and skipped otherwise. This mirrors a similar convention used in `crux-llvm` (see [this code](https://github.com/GaloisInc/crucible/blob/1a7f06e192758c33d614edc8cf4f6fffdc8cea93/crux-llvm/test/Test.hs#L309-L312)), and we could reuse some of the code that it uses to check the versions of tools that you are using.

**Comment by kquick** [comment]
Agreed, and this type of suffix-driven selectability is precisely the type of thing `tasty-sugar` is designed to help with.

**Comment by RyanGlScott** [comment]
How does `tasty-sugar` work for a use case like `disasm-test`'s where there are no golden files? My understanding of how `disasm-test` is that for each `.ll` file, it does the following:

1. Run `llvm-as` on it to produce a bitcode file
2. Parse it back using `llvm-pretty-bc-parser`
3. Pretty-print it using `llvm-pretty`'s pretty-printer
4. Compare the pretty-printed output to the original `.ll` file using `diff` (after normalizing the output)

This process does not involve any golden files, as each `.ll` is compared against itself. I'm unclear on how to make `tasty-sugar` work in this situation. My first inclination was to try something like this:

```hs
cube :: TS.CUBE
cube = TS.mkCUBE { TS.inputDirs = ["disasm-test/tests"]
                 , TS.rootName = "*.ll"
                 , TS.separators = "."
                 , TS.expectedSuffix = "ll"
                 }
```

But this fails to find anything when passed to `findSugar`. I can only assume that this is because `expectedSuffix` clashes with `rootName`—if I change `expectedSuffix` to something else like `.ll-dummy` and copy each `.ll` file with a `.ll-dummy` suffix, then `findSugar` is able to find the `.ll` files. Of course, I really don't want to have to do this...

**Comment by kquick** [comment]
Just released version 2.1.0.0 of tasty-sugar which should support the expected file being the same as the root file.

**Comment by RyanGlScott** [comment]
Thanks for fixing that so quickly, @kquick! With `tasty-sugar-2.1.0.0`, I was able to implement this idea in #210.
